### PR TITLE
Nil message sign verifier for stake

### PR DESCRIPTION
**Pull request by sasurobert**
transactions to interact with staking smart contract. Receiver Address: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l

Stake tx.Data: "stake" + "@" + hex.EncodeToString(big.NewInt(NR_NODES).Bytes()) + "@" + LIST (hex.EncodeToString(BLSPubKey) + "@" + hex.EncodeToString([]byte("msg")) NR_NODES times)  @ hexEncoded(REWARDADDRESS) (the last is optional)

UnStake tx.Data: "unStake" + "@" + LIST(hex.EncodeToString(BLSPubKey)) with "@" separator

UnBond tx.Data: "unBond" + "@" + LIST(hex.EncodeToString(BLSPubKey)) with "@" separator

For UnStake and UnBond only the owner can do it, so the one who called Stake.

UnJail tx.Data: "unJail" + "@" + LIST(hex.EncodeToString(BLSPubKey)) with "@" separator

ChangeRewardAddress: tx.Data: "changeRewardAddress" + "@" + hex.encoded(newRewardAddress)

NodePrice:   500000000000000000000000
UnJailValue: 500000000000000000000

In case of staking, when the message sign verifier is activated instead of hex.EncodeToString([]byte("msg"). You need hex.EncodeToString(Sign TX pubkey with BLS private key)

### CODE REVIEW

File: cmd/node/factory/structs.go [hunk 1]
```diff
@@ -472,9 +473,14 @@ func CryptoComponentsFactory(args *cryptoComponentsFactoryArgs) (*Crypto, error)
 
 	txSignKeyGen := signing.NewKeyGenerator(ed25519.NewEd25519())
 
-	messageSignVerifier, err := systemVM.NewMessageSigVerifier(args.keyGen, singleSigner)
-	if err != nil {
-		return nil, err
+	var messageSignVerifier vm.MessageSignVerifier
```
Comment (SebastianMarian) [inline cmd/node/factory/structs.go]: You can replace this line with: messageSignVerifier := &genesis.NilMessageSignVerifier{} and drop the else branch

File: factory/cryptoComponents.go [hunk 2]
```diff
@@ -87,9 +92,14 @@ func (ccf *cryptoComponentsFactory) Create() (*CryptoComponents, error) {
 
 	txSignKeyGen := signing.NewKeyGenerator(ed25519.NewEd25519())
 
-	messageSignVerifier, err := systemVM.NewMessageSigVerifier(ccf.keyGen, singleSigner)
-	if err != nil {
-		return nil, err
+	var messageSignVerifier vm.MessageSignVerifier
```
Comment (SebastianMarian) [inline factory/cryptoComponents.go:95]: You can replace this line with: messageSignVerifier := &genesis.NilMessageSignVerifier{} and drop the else branch

Comment (sasurobert) [reply factory/cryptoComponents.go:95]: can't do it. as it says type missmatch

File: cmd/node/factory/structs.go (hunk 1 above)
Comment (sasurobert) [reply cmd/node/factory/structs.go]: can't do it. as it says type missmatch

File: vm/systemSmartContracts/auction_test.go [hunk 3]
```diff
@@ -638,7 +640,7 @@ func TestAuctionStakingSC_ExecuteStake(t *testing.T) {
 	retCode := sc.Execute(arguments)
 	assert.Equal(t, vmcommon.Ok, retCode)
 
-	var registrationData StakedData
+	var registrationData AuctionData
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/auction_test.go:643]: :crossed_swords:

File: epochStart/bootstrap/shardStorageHandler.go [hunk 4]
```diff
@@ -261,13 +251,13 @@ func (ssh *shardStorageHandler) saveLastCrossNotarizedHeaders(meta *block.MetaBl
...
 
 	crossNotarizedHdrs := make([]bootstrapStorage.BootstrapHeaderInfo, 0)
-	crossNotarizedHdrs = append(crossNotarizedHdrs, bootstrapStorage.BootstrapHeaderInfo{
-		ShardId: core.MetachainShardId,
-		Nonce:   neededHdr.GetNonce(),
-		Hash:    shardData.LastFinishedMetaBlock,
-	})
-
 	if len(shardData.PendingMiniBlockHeaders) > 0 {
+		crossNotarizedHdrs = append(crossNotarizedHdrs, bootstrapStorage.BootstrapHeaderInfo{
```
Comment (SebastianMarian) [inline epochStart/bootstrap/shardStorageHandler.go:255]: I think that also lines between 238-251 could be moved under this if, but anyway they are completely harmless even if they would stay for now there.

Comment (sasurobert) [reply epochStart/bootstrap/shardStorageHandler.go:255]: done

**Comment by LucianMincu** [comment]
`/node/config` should be renamed named as `/network/config`, as the values returned are network related (general values for interacting with the protocol) and not with the local node.
`/network` should be better named as `/network/status` 

on proxy `/network/status/shardID` && `/network/config`

**Pull request merged**